claude: add DefaultGenerator impl for PathBuf (#215)
hegeldev/hegel-rust: RELEASE.md

RELEASE_TYPE: minor

> This release adds a `DefaultGenerator` impl for `PathBuf`, so `#[derive(Generate)]` works for structs with `PathBuf` fields and `gs::default::<PathBuf>()` returns a generator.

Author: ```suggestion
RELEASE_TYPE: patch

This release implements `DefaultGenerator` for `PathBuf`.
```
